File: app/src/main/java/ml/dvnlabs/animize/ui/fragment/popup/ProfilePop.java
/*
 * Copyright (c) 2020.
 * Animize Devs
 * Copyright 2019 - 2020
 * Davin Alfarizky Putra Basudewa <dbasudewa@gmail.com>
 * This program used for watching anime without ads.
 *
 */

package ml.dvnlabs.animize.ui.fragment.popup;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import ml.dvnlabs.animize.R;
import ml.dvnlabs.animize.database.InitInternalDBHelper;
import ml.dvnlabs.animize.database.model.userland;
import ml.dvnlabs.animize.ui.activity.MainActivity;

public class ProfilePop extends BottomSheetDialogFragment {
    private InitInternalDBHelper initInternalDBHelper;
    private TextView textname,textemail;
    private Button btn_logout;



    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        final View view = inflater.inflate(R.layout.profilepop,container,false);
        view.setClipToOutline(true);
        btn_logout = (Button)view.findViewById(R.id.dash_profile_button_logout);
        textemail = (TextView)view.findViewById(R.id.dash_profile_tv_email);
        textname = (TextView)view.findViewById(R.id.dash_profile_tv_name);


        initInternalDBHelper = new InitInternalDBHelper(getContext());
        SqliteReadUser sqliteReadUser = new SqliteReadUser();
        sqliteReadUser.execute("OK");
        btn_logout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                signout();
            }
        });
        return view;
    }


    public void signout(){
        SqliteLoginBackground sqliteLoginBackground = new SqliteLoginBackground();
        sqliteLoginBackground.execute("SIGNOUT");
    }

    public class SqliteLoginBackground extends AsyncTask<String,Void,String> {
        @Override
        protected void onPreExecute(){

        }
        @Override
        protected String doInBackground(String... params){

            return initInternalDBHelper.signOut();

        }

        @Override
        protected void onPostExecute(String usl){
            Intent intent = new Intent(getContext(), MainActivity.class);
            startActivity(intent);
        }
    }
    public class SqliteReadUser extends AsyncTask<String,Void, userland>{
        @Override
        protected void onPreExecute(){

        }
        @Override
        protected userland doInBackground(String... params){

            return initInternalDBHelper.getUser();

        }

        @Override
        protected void onPostExecute(userland usl){
            textemail.setText(usl.getEmail());
            textname.setText(usl.getNameUser());

        }
    }

}

File: app/src/main/java/ml/dvnlabs/animize/ui/fragment/popup/ProfilePop.kt
/*
 * Copyright (c) 2020.
 * Animize Devs
 * Copyright 2019 - 2020
 * Davin Alfarizky Putra Basudewa <dbasudewa@gmail.com>
 * This program used for watching anime without ads.
 *
 */
package ml.dvnlabs.animize.ui.fragment.popup

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import ml.dvnlabs.animize.R
import ml.dvnlabs.animize.database.AnimizeDatabase
import ml.dvnlabs.animize.database.notification.StarredNotificationDatabase
import ml.dvnlabs.animize.ui.activity.MainActivity
import org.koin.android.ext.android.inject

class ProfilePop : BottomSheetDialogFragment() {
    private val animizeDB: AnimizeDatabase by inject()
    private val recentDB: StarredNotificationDatabase by inject()
    private var textname: TextView? = null
    private var textEmail: TextView? = null
    private var btnLogout: Button? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.profilepop, container, false)
        view.clipToOutline = true
        btnLogout = view.findViewById<View>(R.id.dash_profile_button_logout) as Button
        textEmail = view.findViewById<View>(R.id.dash_profile_tv_email) as TextView
        textname = view.findViewById<View>(R.id.dash_profile_tv_name) as TextView

        GlobalScope.launch {
            readUser()
        }

        btnLogout!!.setOnClickListener { signOut() }
        return view
    }

    private fun signOut() {
        GlobalScope.launch {
            signOutDatabase()
        }
    }

    private suspend fun signOutDatabase() {
        withContext(Dispatchers.IO) {
            animizeDB.animeDAO().deleteAll()
            animizeDB.userDAO().deleteUser()
            recentDB.starredNotificationDAO().deleteALL()
            withContext(Dispatchers.Main) {
                val intent = Intent(context, MainActivity::class.java)
                startActivity(intent)
            }
        }
    }

    private suspend fun readUser() {
        withContext(Dispatchers.IO) {
            val usl = animizeDB.userDAO().getUser()
            withContext(Dispatchers.Main) {
                textEmail!!.text = usl.email
                textname!!.text = usl.nameUser
            }
        }
    }
}